gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/CourseGradeColumnHeaderPanel.java

```diff
@@ -36,16 +41,28 @@ public void onInitialize() {
 
 		final Gradebook gradebook = this.businessService.getGradebook();
 		final GradebookPage gradebookPage = (GradebookPage) getPage();
+		final GbRole role = gradebookPage.getRole();
```

Comment:
Here the role is fetched from the GradebookPage but above its the BasePage. Would be solved with a cache on the service.